cli/azd/pkg/azdo/utils.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+)
+
+// helper method to verify that a configuration exists in the .env file or in system environment variables
+func ensureConfigExists(ctx context.Context, env *environment.Environment, key string, label string) (string, error) {
+	value := env.Values[key]
+	if value != "" {
+		return value, nil
+	}
+
+	value, exists := os.LookupEnv(key)
+	if !exists || value == "" {
+		return value, fmt.Errorf("%s not found in environment variable %s", label, key)
+	}
+	return value, nil
+}
+
+// helper method to ensure an Azure DevOps PAT exists either in .env or system environment variables
+func EnsurePatExists(ctx context.Context, env *environment.Environment, console input.Console) (string, error) {
+	value, err := ensureConfigExists(ctx, env, AzDoPatName, "azure devops personal access token")
+	if err != nil {
+		console.Message(ctx, output.WithWarningFormat("You need an Azure DevOps Personal Access Token (PAT). Please create a PAT by following the instructions here https://aka.ms/azure-dev/azdo-pat"))
+		pat, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message:      "Personal Access Token (PAT):",
+			DefaultValue: "",
+		})
+		if err != nil {
+			return "", fmt.Errorf("asking for pat: %w", err)
+		}
+		// set the pat as an environment variable for this cmd run
+		// note: the scope of this env var is only this shell invocation and won't be available in the caller parent shell
+		os.Setenv(AzDoPatName, pat)
+		value = pat
+
+		idx, err := console.Select(ctx, input.ConsoleOptions{
+			Message: fmt.Sprintf("Would you like to save the PAT to the %s environment file (.env)?", env.GetEnvName()),
```

Comment:
I think we should be moving away from storing credentials in the plaintext `.env` file. An Azure DevOps connection is potentially even more sensitive than a dev environment access key, given that the Azure DevOps likely is multitenanted and has other applications hosted in the same AzDo instance.

Reply:
@ellismg @wbreza I don't know if it's been a set-in-stone design decision, but I think we've avoided storing secrets in `.env` up to this point. I wonder if this is something worthy of a discussion.

Reply:
folks can call `azd env set mysecret xyzsupersecret` :)

There's no way to protect secrets in the environment.  What we don't want is to persist the secret without asking the customer about it.